src/views/webview-app/overview-page.tsx

```diff
@@ -55,30 +61,83 @@ const OverviewPage: React.FC = () => {
     resetGlobalCSS();
   }, []);
 
+  function handleOpenFileChooserResult<T>(): Promise<T> {
+    const requestId = handleOpenFileChooser();
+    return new Promise((resolve) => {
+      const messageHandler = (event) => {
+        const message: MESSAGE_FROM_EXTENSION_TO_WEBVIEW = event.data;
+        if (
+          message.command === MESSAGE_TYPES.OPEN_FILE_CHOOSER_RESULT &&
+          message.requestId === requestId
+        ) {
+          window.removeEventListener('message', messageHandler);
+          resolve(message.files as T);
+        }
+      };
+      window.addEventListener('message', messageHandler);
+    });
+  }
+
+  // Electron 32.0 removed support for the `path` property of the Web File object in favor of the webUtils.getPathForFile method.
+  // https://github.com/electron/electron/blob/83d704009687956fb4b69cb13ab03664d7950118/docs/breaking-changes.md%23removed-filepath
+  // We can not import `dialog` and `webUtils` from 'electron' in the sandboxed webview.
+  // To work around this, we use a custom dialog provider that uses webview APIs
+  // to send a message to the extension process to open the electron file dialog
+  // and listen for the response to get the file path and send them to the electron file input backend.
+  const dialogProvider = {
+    getCurrentWindow(): void {},
+    dialog: {
+      showSaveDialog(): Promise<{ canceled: boolean; filePath?: string }> {
```

Comment:
Wouldn't we want to propagate options passed to this function? Wouldn't we want to propagate the dialog type (save vs open)?